rest/route/createhost_routes.go

```diff
@@ -4,16 +4,85 @@ import (
 	"context"
 	"net/http"
 
+	"github.com/evergreen-ci/evergreen/apimodels"
+	dbModel "github.com/evergreen-ci/evergreen/model"
 	"github.com/evergreen-ci/evergreen/rest"
 	"github.com/evergreen-ci/evergreen/rest/data"
 	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/evergreen/util"
 	"github.com/evergreen-ci/gimlet"
+	"github.com/gorilla/mux"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
 )
 
+type createHostHandler struct {
+	taskID     string
+	createHost apimodels.CreateHost
+}
+
 type listCreateHostHandler struct {
 	taskID string
 }
 
+func getCreateHostRouteManager(route string, version int) *RouteManager {
```

Comment:
implement new hosts as gimlet framework routes?

Reply:
Done, though I'm still using model.Model and BuildFromService. Is that the right thing?